src/core/thread/address_resolver.cpp

```diff
@@ -273,13 +330,15 @@ otError AddressResolver::AddCacheEntry(const Ip6::Address &aEid, Mac::ShortAddre
 
     VerifyOrExit(entry != NULL, error = OT_ERROR_NO_BUFS);
 
-    entry->mTarget   = aEid;
-    entry->mRloc16   = aRloc16;
-    entry->mTimeout  = 0;
-    entry->mFailures = 0;
-    entry->mState    = Cache::kStateCached;
+    entry->mTarget              = aEid;
+    entry->mLastTransactionTime = static_cast<uint32_t>(kLastTransactionTimeInvalid);
+    entry->mRloc16              = aRloc16;
+    entry->mRetryTimeout        = 0;
+    entry->mTimeout             = 0;
+    entry->mFailures            = 0;
+    entry->mState               = Cache::kStateCached;
 
-    MarkCacheEntryAsUsed(*entry);
+    MoveCacheEntryBehindCached(*entry);
```

Comment:
Suggestion: Should we add a log here? We have a log everywhere else address cache entries are changed.
```
otLogNoteArp("Cache entry added (rx inspection): %s, 0x%04x", aEid.ToString().AsCString(), aRloc16);
```